src/main.ts

```diff
@@ -0,0 +1,129 @@
+import os = require('os');
+
+import * as exec from '@actions/exec';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import * as io from '@actions/io';
+import * as toolCache from '@actions/tool-cache';
+import * as rustCore from '@actions-rs/core';
+
+function getPlatformMatchingTarget(): string {
+    const platform = os.platform() as string;
+    switch (platform) {
+        case "linux":
+            return "x86_64-unknown-linux-gnu"
+        case "win32":
+            return "x86_64-pc-windows-msvc"
+        case "darwin":
+            return "x86_64-apple-darwin"
+        default:
+            throw new Error("Unsupported runner");
+    }
+}
+
+function optionIfValueProvided(option: string, value?: string): string {
+    return value ? ` ${option} ${value}` : '';
+}
+
+function getCheckReleaseArguments(): string[] {
+    return [
+        optionIfValueProvided('--package', rustCore.input.getInput('crate-name')),
```

Comment:
Since we're already making breaking changes to the action, this is a great time to change `crate-name` to `package`. That makes it match the underlying option name, and is also cleaner since the package spec is not always just a crate name: https://doc.rust-lang.org/cargo/reference/pkgid-spec.html

Ideally, we'd link to the above in the README when explaining what the `package` does.